### PR TITLE
implemented feature for allowing async inference config

### PR DESCRIPTION
**Pull request by thomasbellio**
## Related Issues/PRs
 
 #7954

## What changes are proposed in this pull request?

In this PR we add a parameter for updating and creating deployments with async inference config

## How is this patch tested?

- [x ] Existing unit/integration tests
- [ x] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [x ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [x ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [x ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ x ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/sagemaker/__init__.py [hunk 1]
```diff
@@ -2336,8 +2366,9 @@ def update_deployment(
                          Defaults to ``None``.
 
                        - ``variant_name``: A string specifying the desired name when creating a
-                                           production variant.  Defaults to ``None``.
-
+                                           production variant.  Defaults to ``None``.                                           
+                       - ``endpoint_config``: A dictionary specifying the endpoint configuration 
+                        as key-value pairs to be set for the deployed endpoint
```
Comment (dbczumar) [inline mlflow/sagemaker/__init__.py:2371]: Nit: Should this be `async_inference_config` instead of `endpoint_config` + a different docstring?

**Review by dbczumar — APPROVED** [review]
LGTM with a tiny docs comment: https://github.com/mlflow/mlflow/pull/8009/files#r1134690730. Excited to merge once it's addressed. Thanks @thomasbell1985 !

File: mlflow/sagemaker/__init__.py (hunk 1 above)
Comment (thomasbell1985) [reply mlflow/sagemaker/__init__.py]: @dbczumar  Good catch, thanks. I have updated it.

**Comment by thomasbell1985** [comment]
> 

Thanks for the approval @dbczumar. Just curious what would be the next steps for getting this one merged, and will a new version be created for installation?

**Comment by dbczumar** [comment]
Thanks @thomasbell1985 ! Merging :). We'll make a new release some time in April. Thank you for your contribution!

**Pull request merged**